project-base/storefront/components/Pages/HomePage/ExchangeTokenHandler.tsx

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+import { useLoginViaExchangeToken } from 'utils/auth/useLoginViaExchangeToken';
+import { showErrorMessage } from 'utils/toasts/showErrorMessage';
+
+export const ExchangeTokenHandler: FC = () => {
+    const router = useRouter();
+    const loginViaExchangeToken = useLoginViaExchangeToken();
+    const processedTokens = useRef(new Set<string>());
+
+    useEffect(() => {
+        const { exchangeToken } = router.query;
```

Comment:
IF (and that's a big if) we would be able to use POST and do not pass the token in url, that would be really nice touch.

Reply:
let's talk about it, please, I am afraid I am not able to do that easily in an acceptable way :smile: